poetry/repositories/pypi_repository.py

```diff
@@ -298,17 +299,22 @@ def _get_release_info(self, name, version):  # type: (str, str) -> dict
                 if "-none-any" not in filename:
                     continue
 
+                # Found a universal wheel, use it
+                urls[url["packagetype"]] = url["url"]
+
             if not urls or "bdist_wheel" not in urls:
-                # If we don't have urls, we try to take the first one
-                # we find and go from there
+                # If we don't have urls, try to find one matching the python version
                 if not json_data["urls"]:
                     return data
 
                 for url in json_data["urls"]:
-                    # Only get sdist and universal wheels if they exist
                     dist_type = url["packagetype"]
+                    python_version = url["python_version"]
+                    v_info = sys.version_info
 
-                    if dist_type != "bdist_wheel":
+                    if dist_type != "bdist_wheel" or python_version != "{}.{}".format(
```

Comment:
This check is not yet correct; eg PyYaml has https://pypi.org/project/PyYAML/#files (cp36 etc…)

Reply:
I talke to Donald Stufft and apparently we should never use python_version:
```
<dstufft> the python_version column is kind of useless
<dstufft> it's just passed along from the uploader
<dstufft> and different uploaders put different things in it
<dstufft> the thing you want to look at for wheels is the actual filename
```
And for parsing the filename information:
```
<dstufft> pip/_internal/wheel.py and pip/_internal/pep426tags.py has what you want there
```